### PR TITLE
Charm application

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: N/A

### Overview

Charm the Django application

### Rationale

N/A

### Module Changes

N/A

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: charm/charmcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    limit: 1
+
+config:
+  options:
+    dns_repository_url:
+      type: string
+      description: The repository URL where the DNS records are stored.
+    secret_key:
+      type: string
+      description: Django secret key.
```
Comment (gregory-schiano) [inline charm/charmcraft.yaml:50]: Isn't secret_key automatically managed by the charm ?

Comment (arturo-seijas) [reply charm/charmcraft.yaml:50]: Not that I'm aware of. Is it @weiiwang01 ?

Comment (gregory-schiano) [reply charm/charmcraft.yaml:50]: Should be: https://github.com/canonical/xiilib/blob/feat-12f-django/xiilib/django/charm.py#L94

Comment (gregory-schiano) [reply charm/charmcraft.yaml:50]: Re-opening, as I think there's a misunderstanding
This is managed by the 12-factor, so unnecessary:
https://github.com/weiiwang01/charmcraft/blob/feat-12f-django/charmcraft/extensions/django.py#L40-L43

The extension will automatically add a config option for it, and the charm will inject it into your workload as "DJANGO_SECRET_KEY" environment variable

Comment (arturo-seijas) [reply charm/charmcraft.yaml]: Ahh. Modified

**Comment by gregory-schiano** [comment]
I think you'll also need to change the environment variable for the `SECRET_KEY` in your application, and use `DJANGO_SECRET_KEY` instead (should be in app/settings.py)

**Comment by arturo-seijas** [comment]
> I think you'll also need to change the environment variable for the `SECRET_KEY` in your application, and use `DJANGO_SECRET_KEY` instead (should be in app/settings.py)

Django is picking up the key as it is now. What do you mean exactly?

**Review by weiiwang01 — DISMISSED** [review]

**Comment by arturo-seijas** [comment]
> > > I think you'll also need to change the environment variable for the `SECRET_KEY` in your application, and use `DJANGO_SECRET_KEY` instead (should be in app/settings.py)
> > 
> > 
> > Django is picking up the key as it is now. What do you mean exactly?
> 
> The application is using `SECRET_KEY` as environment variable to get the secret key: https://github.com/canonical/httprequest-lego-provider/blob/main/app/settings.py#L24
> 
> But the charm doesn't inject `SECRET_KEY` but injects `DJANGO_SECRET_KEY` as environment variable: https://github.com/canonical/xiilib/blob/feat-12f-django/xiilib/django/charm.py#L94
> 
> So either Django automatically fetches `DJANGO_SECRET_KEY` which I don't think is the case, or you need to change the environment variable in `settings.py` to get the one injected by the charm

I think you aren't looking at the last version of the code, the app is using `DJANGO_SECRET_KEY`

**Comment by gregory-schiano** [comment]
> > > > I think you'll also need to change the environment variable for the `SECRET_KEY` in your application, and use `DJANGO_SECRET_KEY` instead (should be in app/settings.py)
> > > 
> > > 
> > > Django is picking up the key as it is now. What do you mean exactly?
> > 
> > 
> > The application is using `SECRET_KEY` as environment variable to get the secret key: https://github.com/canonical/httprequest-lego-provider/blob/main/app/settings.py#L24
> > But the charm doesn't inject `SECRET_KEY` but injects `DJANGO_SECRET_KEY` as environment variable: https://github.com/canonical/xiilib/blob/feat-12f-django/xiilib/django/charm.py#L94
> > So either Django automatically fetches `DJANGO_SECRET_KEY` which I don't think is the case, or you need to change the environment variable in `settings.py` to get the one injected by the charm
> 
> I think you aren't looking at the last version of the code, the app is using `DJANGO_SECRET_KEY`

Indeed, my bad

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by weiiwang01 — APPROVED** [review]

**Pull request merged**